Sources/Core/ObjectiveCIR.swift

```diff
@@ -338,7 +338,29 @@ public struct ObjCIR {
             case let .function(method):
                 return ["\(method.signature);"]
             case let .enumDecl(name, values):
-                return [ObjCIR.enumStmt(name) {
+                let min: Int
+                let max: Int
+                switch values {
+                case let .integer(options):
+                    let values = options.map { $0.defaultValue }
+                    min = values.min() ?? 0
+                    max = values.max() ?? Int.max
+                case let .string(options, _):
+                    min = 0
+                    max = options.count
+                }
+                let enumType: String
+                switch abs(max - min) {
+                case 0 ... Int(UInt8.max):
+                    enumType = min < 0 ? "char" : "unsigned char"
```

Comment:
Is this correct for a enum with the range (-1...Int8.max+1)? It seems like the `abs(max - min)` bit should cover those cases correctly, but I just want to make sure.

Reply:
I don't know; I think I want to abstract this tiny part out so I may build many unit test around it.

Reply:
added tests; i think it looks right. Please take a look!